### PR TITLE
jyg-UID2-1662 remove mocked crypto in tests

### PR DESCRIPTION
**Pull request by jingyi-gao-ttd**
- Instead of stubbing, I utilized the [Node.js Crypto Module](https://www.w3schools.com/nodejs/ref_crypto.asp) to enhance the reliability of cryptographic during testing and enable us to cover all edge cases of API responses as we expand our cryptographic operations.
- Fixed an issue where getAdvertisingTokenAsync would never resolve when the API response could not be decoded. 
- Introduced new test cases to cover scenarios where API responses cannot be decoded
Minor Improvements
- To align with the usage of the Node.js Crypto Module, I have updated the engines field in package.json to specify Node.js version >= 18.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
     "uid2-examples": "webpack --mode=development --env outputToExamples=true",
     "build:esp": "webpack --env espOnly=true"
   },
+  "engines": {
+    "node": ">=18"
```
Comment (lionell-pack-ttd) [inline package.json:24]: Node versions <18 are out of security support, so this seems like a good idea.

File: src/integrationTests/basic.test.ts [hunk 2]
```diff
@@ -529,9 +539,17 @@ testCookieAndLocalStorage(() => {
...
-        xhrMock.responseText = btoa(JSON.stringify({ status: "optout" }));
-        xhrMock.onreadystatechange(new Event(""));
+      beforeEach(async () => {
+        try {
+          advertisingTokenRefreshedPromise = uid2.getAdvertisingTokenAsync();
+          await xhrMock.sendEncodedRefreshApiResponse(
+            "optout",
+            originalIdentity.refresh_response_key
+          );
+          await advertisingTokenRefreshedPromise;
```
Comment (lionell-pack-ttd) [inline src/integrationTests/basic.test.ts:549]: I've seen this pattern in a lot of tests - why do we need to do this now, when we didn't before?

File: src/integrationTests/autoRefresh.test.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ beforeEach(() => {
```
Comment (alex-yau-ttd) [inline src/integrationTests/autoRefresh.test.ts:23]: I think we can now remove `_cryptoMock` in this file?

Comment (alex-yau-ttd) [reply src/integrationTests/autoRefresh.test.ts:23]: wait nevermind, got confused

File: src/integrationTests/callbacks.test.ts [hunk 4]
```diff
@@ -131,29 +128,25 @@ testCookieAndLocalStorage(() => {
           identity,
         });
       });
-      // this test is only passing for cookies
-      test('it should receive subsequent identity updates', async () => {
+
+      test('it should receive subsequent identity updates', (done) => {
```
Comment (alex-yau-ttd) [inline src/integrationTests/callbacks.test.ts:132]: This is my first time seeing `done` - could you please help me understand why we need it here? I can see it improves the coverage of the test.

Comment (jingyi-gao-ttd) [reply src/integrationTests/callbacks.test.ts:132]: `done` is used only when you want to test a callback, Jest will wait until the done callback is called before finishing the test.
https://jestjs.io/docs/asynchronous#callbacks

**Review by alex-yau-ttd — APPROVED** [review]
